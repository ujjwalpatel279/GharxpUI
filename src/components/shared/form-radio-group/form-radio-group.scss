// ========================================================================
// COMPONENTS/SHARED/FORMS/RADIO-GROUP
// ========================================================================

@import '~sassHelpers';

.c-radio-input {
  position: relative;
  display: inline-block;
  overflow: hidden;
  cursor: pointer;

  &__inner {
    display: flex;
    align-items: center;
    justify-content: center;
    height: spacer(10);
    padding-left: spacer(3);
    padding-right: spacer(3);
    background-color: white;
    border: 1px solid color-brand(secondary);
    border-radius: $border-radius;
  }

  input {
    position: absolute;
    top: -100px;
    left: -100px;

    &:focus ~ .c-radio-input__inner {
      outline: 1px dashed color-brand(secondary);
      outline-offset: -4px;
    }

    &:checked ~ .c-radio-input__inner {
      background-color: color-brand(secondary);
      color: white;
    }
  }
}
