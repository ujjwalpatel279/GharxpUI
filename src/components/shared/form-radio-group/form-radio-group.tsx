import React, { forwardRef } from 'react';

import './form-radio-group.scss';

interface Attributes {
  id: number;
  value: string;
}

interface FormRadioGroupProps {
  name: string;
  attributes: Attributes[];
}

export const FormRadioGroup = forwardRef(
  ({ name, attributes }: FormRadioGroupProps, ref: React.Ref<HTMLInputElement>) => (
    <div className="l-flex l-flex--wrap">
      {attributes.map((attribute: Attributes) => (
        <div key={attribute.id}>
          <label htmlFor={attribute.value} className="c-radio-input">
            <input type="radio" id={attribute.value} name={name} value={attribute.value} ref={ref} />
            <span className="c-radio-input__inner">
              <span>{attribute.value}</span>
            </span>
          </label>
        </div>
      ))}
    </div>
  ),
);

FormRadioGroup.displayName = 'FormRadioGroup';
