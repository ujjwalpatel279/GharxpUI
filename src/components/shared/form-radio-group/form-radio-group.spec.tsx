import React from 'react';
import { render, cleanup } from '@testing-library/react';

import { FormRadioGroup } from './form-radio-group';

describe('FormRadioGroup Component', () => {
  afterEach(cleanup);

  it('renders the form radio group component', () => {
    const { container } = render(
      <FormRadioGroup name="testName" attributes={[{ id: 1, value: 'test' }]} ref={() => null} />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
