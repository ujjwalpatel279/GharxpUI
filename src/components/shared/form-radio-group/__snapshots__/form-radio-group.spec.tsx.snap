// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormRadioGroup Component renders the form radio group component 1`] = `
<div
  class="l-flex l-flex--wrap"
>
  <div>
    <label
      class="c-radio-input"
      for="test"
    >
      <input
        id="test"
        name="testName"
        type="radio"
        value="test"
      />
      <span
        class="c-radio-input__inner"
      >
        <span>
          test
        </span>
      </span>
    </label>
  </div>
</div>
`;
